engine/src/state_chain_observer/sc_observer.rs

```diff
@@ -417,6 +418,25 @@ where
                                             let btc_rpc = btc_rpc.clone();
                                             let state_chain_client = state_chain_client.clone();
                                             scope.spawn(async move {
+
+                                                // We check for AwaitingBroadcasts for Bitcoin specifically because if the previous broadcast was not broadcast,
+                                                // it can cause ours to fail, as we could be using a change UTXO that's only created in the previous broadcast.
+                                                for (awaiting_broadcast_id, awaiting_broadcast_data) in state_chain_client
+                                                    .storage_map::<pallet_cf_broadcast::AwaitingBroadcast<Runtime, BitcoinInstance>, Vec<_>>(current_block.hash)
+                                                    .await?
+                                                    .into_iter()
```

Comment:
I think AwaitingBroadcast might include aborted broadcasts. 

Better to iterate over PendingBroadcasts and then fetch the information from AwaitingBroadcasts by broadcast+id.